packages/hardhat-viem/test/chains.ts

```diff
@@ -0,0 +1,138 @@
+import type { EthereumProvider } from "hardhat/types";
+
+import { expect, assert } from "chai";
+import sinon from "sinon";
+import * as chains from "viem/chains";
+
+import {
+  getChain,
+  getMode,
+  isDevelopmentNetwork,
+} from "../src/internal/chains";
+import { EthereumMockedProvider } from "./mocks/provider";
+
+describe("chains", () => {
+  describe("getChain", () => {
+    afterEach(sinon.restore);
+
+    it("should return the chain corresponding to the chain id", async () => {
+      const provider: EthereumProvider = new EthereumMockedProvider();
+      const sendStub = sinon.stub(provider, "send");
+      sendStub.withArgs("eth_chainId").returns(Promise.resolve("0x1")); // mainnet chain id
```

Comment:
We should somehow assert that this is the only JSON-RPC call done, shouldn't we? That would help prevent performance regressions where we do unnecessary http requests.

I think we just need to add an `assert.equal(sendStub.callCount, 1);` statement.

I realize that that would make these tests even more whitebox, but I think it's worth it in this case since we are asserting certain guarantees (a minimal amount of requests) that we want to preserve.